src/components/select/list-action-button/list-action-button.spec.tsx

```diff
@@ -1,20 +1,35 @@
 import React, { useRef } from "react";
 import TestRenderer from "react-test-renderer";
-import { shallow, mount } from "enzyme";
-import ListActionButton from "./list-action-button.component";
+import { shallow, mount, ShallowWrapper } from "enzyme";
+import ListActionButton, { ListActionButtonProps } from ".";
 import Button from "../../button";
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function renderListActionButton(
+  props: Partial<ListActionButtonProps>,
+  renderer: any = shallow
+) {
+  const componentProps = { ...props };
+  if (!componentProps.onListAction) {
+    componentProps.onListAction = () => {};
+  }
+
+  return renderer(
+    <ListActionButton {...(componentProps as ListActionButtonProps)} />
+  );
+}
```

Comment:
Apologies, this now causes a knock on effect for line 22 `expect(renderListActionButton({}, TestRenderer.create)).toMatchSnapshot();` as this is using react-test-renderer rather than Enzyme. This is the issue with using `:any` as it's flagging this up now. I can look at how we can update this further if it seems worthwhile or we can ignore for now as it will all be replaced by RTL in the future I guess.

Reply:
As you observe, we do this in pretty much all tests that are written like this, and it's because the `renderer` can typically be either `shallow` or `mount`, and for some reason there doesn't seem to be a way of typing that other than using `any`. (`typeof shallow | typeof mount` seems to not work, although I don't understand why.) Obviously here with using react-test-renderer as well there are probably even more issues.

I'm inclined to ignore it as we do this in many places, and as you say it will go away when we do the RTL refactor anyway - we'll be doing that in TS from the start so hopefully won't be so accepting of `any`.

However looking at your comment has made be realise that prettier has reformatted the code so that the `ignore-next-line` isn't actually ignoring the correct line any more! I'm surprised this isn't causing a TS error...

Reply:
Good spot. And yeah it seems fair enough to be honest, particularly with the "imminent" refactor.